### PR TITLE
Add option to specify number of validation days to LightGBM

### PR DESCRIPTION
**Pull request by rusane**
You can now specify the number of validation days (`default=0`). Also got rid of `LAST_DAY` and `FIRST_DATE` global constants,  instead they are now computed dynamically from the dataframes.

We probably need to run commands like these for the validation `[d_1886, d_1913]` and public test (`[d_1914, d_1941]`) runs:

**Validation**
`python main.py ../kaggle/input/m5-forecasting-accuracy -d 1000 -v 28 -i 500 -s ../models/lgb_model_val.lgb -k`

**Public test set**
`python main.py ../kaggle/input/m5-forecasting-accuracy -d 1000 -i 500 -s ../models/lgb_model_test.lgb -k`

The argument `-v 28` reserves the last 28 days from the data by removing them from the train data. For the public test set run we don't need to remove any days for validation.

We can up the iterations `-i` of course depending on how long we want to train for.

### CODE REVIEW

**Comment by nnistelrooij** [comment]
LGTM

**Review by MauiGonzo — APPROVED** [review]
